src/modules/keepalive/keepalive_rpc.c

```diff
@@ -87,3 +100,137 @@ static void keepalive_rpc_list(rpc_t *rpc, void *ctx)
+		return;
+	}
+
+	if(ka_add_dest(&sip_adress,&table_name,0,0,0) < 0 ){
+		LM_ERR("couldn't add data to list \n"  );
+		rpc->fault(ctx, 500, "couldn't add data to list");
+		return;
+	}
+
+	return;
+}
+static const char *keepalive_rpc_add_doc[2] = {
+		"adds new destination to keepalive memory. usage: keepalive.add sip:username@domain listname", 0};
+
+static void keepalive_rpc_del(rpc_t *rpc, void *ctx)
+{
+	str sip_adress = {0,0};
+	str table_name ={0,0};
+	int ret = 0;
+
+	ret = rpc->scan(ctx, "SS",&sip_adress,&table_name);
+
+	if (ret < 2) {
+		LM_ERR("not enough parameters - read so far: %d\n", ret);
+		rpc->fault(ctx, 500, "Not enough parameters or wrong format");
+		return;
+	}
+
+	LM_DBG(" keepalive deletes [%.*s]\n", sip_adress.len , sip_adress.s);
+
+	if(sip_adress.len < 1 || table_name.len < 1){
+		LM_ERR("parameter is len less than 1  \n");
+		rpc->fault(ctx, 500, "parameter is len less than 1");
+		return;
+	}
+
+	if(ka_del_destination(&sip_adress,&table_name) < 0 ){
+		LM_ERR("couldn't delete data from list \n"  );
+		rpc->fault(ctx, 500, "couldn't delete data from list");
+		return;
+	}
+
+	return;
+}
+static const char *keepalive_rpc_del_doc[2] = {
+		"deletes destination from keepalive memory. usage: keepalive.del sip:username@domain listname", 0};
+
+static void keepalive_rpc_get(rpc_t *rpc, void *ctx)
+{
+	str sip_adress = {0,0};
+	str table_name ={0,0};
+	int ret = 0;
+	ka_dest_t *target=0 , *head =0;
+	void *sub;
+
+	ret = rpc->scan(ctx, "SS",&sip_adress,&table_name);
+
+	if (ret < 2) {
+		LM_ERR("not enough parameters - read so far: %d\n", ret);
+		rpc->fault(ctx, 500, "Not enough parameters or wrong format");
+		return;
+	}
+
+	LM_DBG(" keepalive gets [%.*s]\n", sip_adress.len , sip_adress.s);
+
+	if(sip_adress.len < 1 || table_name.len < 1){
+		LM_ERR("parameter is len less than 1  \n");
+		rpc->fault(ctx, 500, "parameter is len less than 1");
+		return;
+	}
+	ka_lock_destination_list();
+
+	if(ka_find_destination(&sip_adress, &table_name, &target, &head) < 0 ){
+		LM_ERR("couldn't get data from list \n"  );
+		rpc->fault(ctx, 500, "couldn't get data from list");
+		ka_unlock_destination_list();
+
+		return;
+	}
+
+	if(!target){
+		LM_ERR("Target is empty \n"  );
+		rpc->fault(ctx, 500, "couldn't get data from list");
+		ka_unlock_destination_list();
+		return;
+	}
+
+	rpc->add(ctx, "{", &sub);
+
+	rpc->struct_add(sub, "SSd", "uri", &target->uri, "owner", &target->owner,"state", target->state);
+
+	ka_unlock_destination_list();
+
+	return;
+}
+static const char *keepalive_rpc_get_doc[2] = {
+		"gets destination info data  from keepalive memory. usage keepalive.get sip:xx@domain listname", 0};
+
+
+static void keepalive_rpc_flush(rpc_t *rpc, void *ctx)
+{
+	ka_dest_t *dest;
+	LM_DBG("Keepalive flushes  \n");
+	ka_lock_destination_list();
+
+	for(dest = ka_destinations_list->first; dest != NULL; dest = dest->next) {
+		free_destination(dest);
+	}
+	ka_destinations_list->first = 0;
```

Comment:
Are you sure you want to set it to "0" here? first seems to be of type ka_dest_t * so maybe NULL is better?

Reply:
There is no strict policy on using `NULL`  or `0` for pointers, probably the former is more intuitive for pointers and eventually raise compile warnings if the type is not pointer, but there are many places in the code where `0` is used for this purpose. If someone wants to make it coherent everywhere by using `NULL` instead of `0` for pointers, I am fine with it.

Given that the PR is rather old here, I am going to merge it and fix what was pointed in the other comments.